src/Bicep.Core/Semantics/Namespaces/SystemNamespaceSymbol.cs

```diff
@@ -389,6 +391,67 @@ public class SystemNamespaceSymbol : NamespaceSymbol
+                .Build(),
+
+            new FunctionOverloadBuilder("loadContent")
+                .WithReturnType(LanguageConstants.String)
+                .WithDescription("Load content of specified file into a string. Content loading occurs during compilation, not in runtime. Maximum allowed content size is 24,576 characters (including line endings).")
+                .WithRequiredParameter("filePath", LanguageConstants.String, "File path which content will be loaded")
+                .WithOptionalParameter("encoding", LanguageConstants.String, "File encoding. If not provided, Bicep will try to automatically detect the encoding of a file based on the presence of byte order marks.")
+                .WithDiagnosticsCollector((binder, diagnostics, arguments) => {
+                    if (arguments[0].syntax.Expression is StringSyntax filePathSyntax)
+                    {
+                        if (filePathSyntax.IsInterpolated())
+                        {
+                            diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[0].syntax).CompileTimeConstantRequired());
+                            return;
+                        }
+
+                        var filePath = filePathSyntax.TryGetLiteralValue() ?? string.Empty;
+                        if (!SyntaxTreeGroupingBuilder.ValidateFilePath(filePath, out var validateFilePathFailureBuilder))
+                        {
+                            diagnostics.Write(validateFilePathFailureBuilder.Invoke(DiagnosticBuilder.ForPosition(arguments[0].syntax)));
+                            return;
+                        }
+
+                        var fileUri = binder.FileResolver.TryResolveModulePath(binder.FileUri, filePath);
+                        if (fileUri is null)
+                        {
+                            diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[0].syntax).FilePathCouldNotBeResolved(filePath, binder.FileUri.LocalPath));
+                            return;
+                        }
+
+                        if (!fileUri.IsFile)
+                        {
+                            diagnostics.Write(DiagnosticBuilder.ForPosition(arguments[0].syntax).UnableToLoadNonFileUri(fileUri));
+                            return;
+                        }
+
+
+                        Encoding? fileEncoding = null;
+                        if (arguments.Length > 1)
+                        {
+                            fileEncoding = Encoding.UTF8; //todo read encoding
+                        }                        
+
+                        if (!binder.FileResolver.TryRead(fileUri, out var fileContent, out var fileReadFailureBuilder, fileEncoding))
+                        {
+                            diagnostics.Write(fileReadFailureBuilder.Invoke(DiagnosticBuilder.ForPosition(arguments[0].syntax)));
+                            return;
+                        }
+
+                        if (fileContent.Length > 24576)
```

Comment:
>fileContent [](start = 28, length = 11)

A couple thoughs here:
1. Is it possible to check the size/length before reading the contents? Could do it via `FileInfo` or by streaming in the contents into a string up the specified size.
2. How was the limit determined? (Definitely need a limit.)
3. Can we put the limit in a constant?

Reply:
definitely limit will go to a constant. It's taken from here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-best-practices#template-limits.

Also I thought for a larger files that we could split into chunks that will be ARM variables starting with "!" sign (as we cannot use that sign in bicep, but is allowed in ARM) and then do a concat function to combine them.

I could leverage FileInfo to open a text stream with given encoding and try to get total file size, or we can just limit to a bytes

However for API-M openAPI JSONs the limit of 25k chars might not be enough but I'm not sure if there are any limits on ARM side, can you check?

Reply:
I forgot about that limit, but I don't think it applies to variable values that aren't expressions (hard-coded string, for example). If that's true, it would mean we wouldn't need to split the files into chunks - one variable might be sufficient.

I can't think of other limits right now.

What's "API-M"?

Reply:
Api Management :)

Reply:
@majastrz just did some large text check and there's a limit:
`Unhandled exception. Azure.Deployments.Expression.Exceptions.ExpressionException: The language expression literal limit exceeded. Limit: '131072' and actual: '1006448'.`